tests/server/api/routers/v1/test_datasets.py

```diff
@@ -536,3 +537,12 @@ async def test_post_dataset_upload_pyarrow_create_then_append(test_client, sessi
     assert revisions[1].input == {"a": 11, "b": 22, "c": 33}
     assert revisions[1].output == {"b": 22, "c": 33, "d": 44}
     assert revisions[1].metadata_ == {"c": 33, "d": 44, "e": 55}
+
+
+async def test_delete_dataset(test_client, empty_dataset) -> None:
+    url = f"v1/datasets/{GlobalID(Dataset.__name__, str(1))}"
+    assert len((await test_client.get("/v1/datasets")).json()["data"]) > 0
+    (await test_client.delete(url)).raise_for_status()
```

Comment:
Suggestion: Test to ensure the deleted dataset id is returned.

Reply:
looks like it should be 204 https://www.vinaysahni.com/best-practices-for-a-pragmatic-restful-api#:~:text=Response%20to%20a%20successful%20request%20that%20won%27t%20be%20returning%20a%20body%20(like%20a%20DELETE%20request)

> 204 No Content - Response to a successful request that won't be returning a body (like a DELETE request)